models/users.js

```diff
@@ -154,5 +158,58 @@ module.exports = {
     }
     const res = await db.query(auth)
     return (res.rows.length !== 0)
+  },
+
+  getMergedUserId: async function (userId) {
+    const getMergeUserId = {
+      text: 'SELECT mergeduserid FROM users WHERE id = $1',
+      values: [userId]
+    }
+    const res = await db.query(getMergeUserId)
+    return res.rows[0].mergeduserid
+  },
+
+  mergeAccounts: async function (mergeUserId, ogUserId) {
+    const merge = {
+      text: 'UPDATE users SET mergeduserid = $2 WHERE id = $1',
+      values: [mergeUserId, ogUserId]
+    }
+    return db.query(merge)
+  },
+
+  mergeChildAccounts: async function (mergeUserId, ogUserId) {
+    const merge = {
+      text: 'UPDATE users SET mergeduserid = $2 WHERE mergeduserid = $1',
+      values: [mergeUserId, ogUserId]
+    }
+    return db.query(merge)
+  },
+
+  mergeFriends: async function (mergeUserId, ogUserId) {
+    const merge = {
+      text: 'UPDATE friendships SET user1 = $2 WHERE user1 = $1',
+      values: [mergeUserId, ogUserId]
+    }
+    return db.query(merge)
+  },
+
+  removeDuplicateFriends: async function (userId) {
+    const removeDup = {
+      text: 'DELETE FROM friendships a WHERE a.ctid <> ' +
+      '(SELECT min(b.ctid) FROM friendships b WHERE  a.user2 = b.user2 AND a.user1 = $1 AND b.user1 = $1)',
+      values: [userId]
+    }
```

Comment:
wat

Reply:
Honestly, this is mostly boiler plate code. It basically removes all duplicates but 1.